### PR TITLE
Initial merge with refactored ModelSpec from ml-tools

### PR DESCRIPTION
**Pull request by snormore**
Adds:
 - `preprocess_func` and `preprocess_args`
 - `model_specs.json` as part of the package containing base architecture specs

### CODE REVIEW

File: keras_model_specs/model_specs.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "inception_v3": {
+    "klass": "keras.applications.inception_v3.InceptionV3",
```
Comment (stephensolis) [inline keras_model_specs/model_specs.json]: Is `klass` useful at all at this point?

Comment (snormore) [reply keras_model_specs/model_specs.json]: We use it in the `keras-trainer` repo https://github.com/triagemd/keras-trainer/blob/914547976ba20ec0bf4db9af8670486cf7b846b2/keras_trainer/trainer.py#L74

Comment (snormore) [reply keras_model_specs/model_specs.json]: I think the pattern for now is that you want to first choose the base spec from one of these, that determines the class/layers in training, and the target size and preprocessing logic in eval/val. We're using custom target size pretty often, but the core of the others are the same with a little config through params.

Comment (stephensolis) [reply keras_model_specs/model_specs.json]: I'm just thinking towards the future, like with https://github.com/triagemd/keras-trainer/pull/1#discussion_r136431375 -- I guess I'm just not so optimistic how far the stuff in `keras.applications`, unmodified, is going to get us in the medium-to-long term...

But this makes sense for now, we can always iterate later once training patterns become more clear.

**Pull request merged**